src/core/execute.c

```diff
@@ -2002,27 +2005,14 @@ int exec_shared_runtime_deserialize_compat(Unit *u, const char *key, const char
                         return -ENOMEM;
 
         } else if (streq(key, "netns-socket-0")) {
-                int fd;
-
-                if ((fd = parse_fd(value)) < 0 || !fdset_contains(fds, fd)) {
-                        log_unit_debug(u, "Failed to parse netns socket value: %s", value);
-                        return 0;
-                }
 
                 safe_close(rt->netns_storage_socket[0]);
-                rt->netns_storage_socket[0] = fdset_remove(fds, fd);
+                rt->netns_storage_socket[0] = deserialize_fd(fds, value);
```

Comment:
here and below there used to be checks on the parsing, and it returned 0 on failure, but now it just carries on